ckan/templates/error_document_template.html

```diff
@@ -10,6 +10,13 @@ <h1>{{ code }} {{ name }}</h1>
       {% endif %}
       {{ content}}
     </div>
+    {% block login_redirect %}
+    {% if current_user.is_anonymous %}
+    <div class="module-content">
```

Comment:
This message should only be shown in the error page if the raised error was a 403 or 401, not when the user is anonymous. So keep the following check in the `error_handler()` and if it matches you can pass an extra `show_login_redirect_link` var to show this block:

```python
                    if current_user.is_anonymous and type(e) in (
                        Unauthorized, Forbidden
                    ):
```

Also the text needs to be translatable so please use something along these lines:
```
{{ _("You might need to login to access this page.") }} <a href="{{ h.url_for('user.login') }}">{{ _("Click here to login") }}</a> 
```

Reply:
thanks for this, it looks better, you can check my changes :) @amercader